Graphs/Table/index.js

```diff
@@ -93,20 +95,58 @@ class Table extends AbstractGraph {
         } = props
 
         const {
-            selectColumnOption
+            selectColumnOption,
+            matchingRowColumn
         } = this.getConfiguredProperties()
 
         let columns = this.getColumns();
 
         if (!columns)
             return;
+
+        let columnNameList = []
+        this.keyColumns    = []
+        this.filterData    = []
+
+        // generate random key for each column and assign that key to the values in data
+        columns.forEach( d => {
+            columnNameList.push(d.column)
+            this.keyColumns[ d.selection ? d.label : `${d.column}_${Math.floor(100000 + Math.random() * 900000)}`] = d
+        })
+
+        props.data.forEach( d => {
+            let data = {};
+            for(let key in this.keyColumns) {
+                if(this.keyColumns.hasOwnProperty(key)) {
```

Comment:
why is this check required? key comes from this.keyColumns

Reply:
`this.keyColumns.hasOwnProperty(key)` is required because if we don't put this check in the object loop then it will throw a warning in the console and Object.hasOwnProperty determines if the whole property is defined in the object itself or in the prototype chain, and is a standard approach that we follow.